### PR TITLE
EAGLE-79 Provide basic aggregation

### PR DESCRIPTION
**Pull request by RalphSu**
https://issues.apache.org/jira/browse/EAGLE-79

1. Support aggregation api
2. Support persist Api
3. Refactor AlertExecutor to be policy wide.

### CODE REVIEW

File: eagle-core/eagle-data-process/eagle-stream-process-api/src/main/java/org/apache/eagle/dataproc/impl/persist/druid/DruidPersistService.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+import java.util.*;
+import java.util.concurrent.Future;
+
+/**
+ * TODO : support more general entity input
+ * @since Dec 21, 2015
+ *
+ */
+public class DruidPersistService implements IPersistService<AggregateEntity> {
```
Comment (haoch) [inline eagle-core/eagle-data-process/eagle-stream-process-api/src/main/java/org/apache/eagle/dataproc/impl/persist/druid/DruidPersistService.java]: How do you think to rename the `DruidPersistService` to `KafkaPersistService` here as it's for kafka only in fact. We may have another `DruidPersistService` using https://github.com/druid-io/tranquility later.

File: eagle-core/eagle-policy/eagle-policy-base/pom.xml [hunk 2]
```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0"?>
```
Comment (haoch) [inline eagle-core/eagle-policy/eagle-policy-base/pom.xml]: It's good to separate policy engine out for reusing in both alerting and aggregation.

File: eagle-core/eagle-policy/eagle-policy-base/src/test/java/org/apache/eagle/policy/dao/TestSchemaDao.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import java.util.List;
+
+/**
+ * Created on 12/31/15.
+ */
+public class TestSchemaDao {
+
+    @Test
+    public void test() throws Exception {
+        AlertStreamSchemaDAO dao = new AlertStreamSchemaDAOImpl("eagle-c3-lvs01-1-9953.lvs01.dev.ebayc3.com", 9099, "admin", "secret");
```
Comment (haoch) [inline eagle-core/eagle-policy/eagle-policy-base/src/test/java/org/apache/eagle/policy/dao/TestSchemaDao.java]: Please clean sensitive info

File: eagle-samples/eagle-persist-sample/pom.xml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>eagle</groupId>
+            <artifactId>eagle-stream-process-api</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>eagle</groupId>
+            <artifactId>eagle-query-base</artifactId>
+            <version>0.3.0</version>
```
Comment (haoch) [inline eagle-samples/eagle-persist-sample/pom.xml]: Please use ${project.version} to specify module version if not have to.

**Comment by haoch** [comment]
Have you implemented the inline sql-based-policy interface?

File: eagle-core/eagle-data-process/eagle-stream-process-api/src/main/java/org/apache/eagle/dataproc/impl/persist/druid/DruidPersistService.java (hunk 1 above)
Comment (RalphSu) [inline eagle-core/eagle-data-process/eagle-stream-process-api/src/main/java/org/apache/eagle/dataproc/impl/persist/druid/DruidPersistService.java]: Nice suggestion, kafakpersistservice seems better.

File: eagle-core/eagle-policy/eagle-policy-base/pom.xml (hunk 2 above)
Comment (RalphSu) [inline eagle-core/eagle-policy/eagle-policy-base/pom.xml]: hm... what does this for?

File: eagle-core/eagle-policy/eagle-policy-base/src/test/java/org/apache/eagle/policy/dao/TestSchemaDao.java (hunk 3 above)
Comment (RalphSu) [inline eagle-core/eagle-policy/eagle-policy-base/src/test/java/org/apache/eagle/policy/dao/TestSchemaDao.java]: Removed

File: eagle-samples/eagle-persist-sample/pom.xml (hunk 4 above)
Comment (RalphSu) [inline eagle-samples/eagle-persist-sample/pom.xml]: Missed ide generation, fixed.

**Comment by RalphSu** [comment]
For sql-based-policy, i think it mean an policy executor accept a fixed sql. This would be added later, not in this commit.

File: eagle-core/eagle-policy/eagle-policy-base/pom.xml (hunk 2 above)
Comment (haoch) [inline eagle-core/eagle-policy/eagle-policy-base/pom.xml]: Never mind, I mistake something. :-)

**Comment by haoch** [comment]
Ok, i think the pull request is good for me.

**Comment by sunlibin** [comment]
The PR looks good to me

**Comment by RalphSu** [comment]
Add aggregate purely on single user given QL support

**Comment by haoch** [comment]
LGTM :+1:

**Comment by haoch** [comment]
Will merge this pull request into master branch today if no additional comments.

**Comment by haoch** [comment]
@RalphSu thanks for the contribution and pull request has been merged, please resolve the JIRA ticket https://issues.apache.org/jira/browse/EAGLE-79.

**Comment by haoch** [comment]
Got RAT exception, but it doesn't matter, i will fix it.

    [ERROR] Failed to execute goal org.apache.rat:apache-rat-plugin:0.11:check (header-check) on project eagle-policy-base: Too many files with unapproved license: 1 See RAT report in: /Users/hchen9/Workspace/incubator-eagle/eagle-core/eagle-policy/eagle-policy-base/target/rat.txt -> [Help 1]

Unapproved licenses:

    incubator-eagle/eagle-core/eagle-policy/eagle-policy-base/src/main/resources/eagle.siddhiext

**Pull request closed without merging**